sdk/core/src/services/github/client/mod.rs

```diff
@@ -0,0 +1,284 @@
+use std::convert::TryFrom;
+use std::path::Path;
+
+use git2::Repository;
+use serde;
+use serde::{Deserialize, Serialize};
+use tracing::info;
+
+use platform::hyper::{
+    ContentType,
+    Error as HyperError,
```

Comment:
This causes a lot of refactor because I transformed Hyper errors for everything here. 

One of the Error enums, `HyperError::Remote(status, msg)`, I actually had to modify (4 times actually - same change due to having to port the code from one branch to the next) to give me back the error code.  To fix this properly, I'd need to start with the Hyper Errors and change the Platform Errors to give me the error code data, then re-code the GitHub specific errors.  I'd love to do that, but can it wait until a subsequent PR? 

```
error[E0433]: failed to resolve: use of undeclared type `HyperError`
  --> sdk/core/src/services/github/client/mod.rs:91:24
   |
91 |                 if let HyperError::Remote(status, msg) = err {
   |                        ^^^^^^^^^^ use of undeclared type `HyperError`

error[E0412]: cannot find type `HyperError` in this scope
  --> sdk/core/src/services/github/client/mod.rs:44:43
   |
44 |         let response: Result<Option<Org>, HyperError> = self.http_client.get(
   |                                           ^^^^^^^^^^ not found in this scope
   |
help: you might be missing a type parameter
   |
27 | impl<HyperError> Client {
   |     ++++++++++++

error[E0412]: cannot find type `HyperError` in this scope
  --> sdk/core/src/services/github/client/mod.rs:74:46
   |
74 |         let response: Result<Option<Commit>, HyperError> = self.http_client.get(
   |                                              ^^^^^^^^^^ not found in this scope
   |
help: you might be missing a type parameter
   |
27 | impl<HyperError> Client {
   |     ++++++++++++

error[E0412]: cannot find type `HyperError` in this scope
   --> sdk/core/src/services/github/client/mod.rs:149:49
    |
149 |         let response: Result<Option<Vec<Repo>>, HyperError> = self.http_client.get(
    |                                                 ^^^^^^^^^^ not found in this scope
    |
help: you might be missing a type parameter
    |
27  | impl<HyperError> Client {
    |     ++++++++++++

^C  Building [=======================> ] 405/414: harbcore
```

Reply:
Yes totally.  Not a blocker.